--- watt3210/src/tcp_fsm.c	2005-10-20 17:04:18.000000000 +0200
+++ watt32/src/tcp_fsm.c	2020-01-21 23:50:06.669660137 +0100
@@ -1072,6 +1072,15 @@
 
   if (ldiff >= 0)
   {
+    /* added 2020-01-21 by Mateusz Viste:
+     * if ldiff > len then data_in_order() will call memcpy() with a negative
+     * length - and disaster will happen. I detect this condition here and
+     * abort operation. this is NOT a solution. I do NOT know what I'm doing,
+     * this is only a DIRTY hack so watt-32 does not crash my PC. Formal
+     * bug report here: https://github.com/gvanem/Watt-32/issues/2          */
+    if (ldiff > len) {
+      return(-1);
+    }
     data_in_order (s, data, len, ldiff);
     s->unhappy = (s->tx_datalen > 0);
     return (0);
